mlflow/transformers.py

```diff
@@ -0,0 +1,851 @@
+        return transformers.pipeline(**components)
+    elif return_type == "components":
+        return components
+    else:
+        raise MlflowException(
+            f"The specified return_type mode '{return_type}' is unsupported. "
+            "Please select one of: 'pipeline' or 'components'.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+
+def _load_model(path: str, flavor_config):
+    """
+    Loads components from a locally serialized ``Pipeline`` object.
+    """
+    import transformers
+
+    local_path = pathlib.Path(path)
+    pipeline_path = local_path.joinpath(
+        flavor_config.get(_PIPELINE_BINARY_KEY, _PIPELINE_BINARY_FILE_NAME)
+    )
+
+    model_instance = getattr(transformers, flavor_config[_PIPELINE_MODEL_TYPE_KEY])
+    conf = {
+        "task": flavor_config[_TASK_KEY],
+        "model": model_instance.from_pretrained(pipeline_path),
+    }
+
+    if flavor_config.get(_PROCESSOR_TYPE_KEY, None):
+        conf[_PROCESSOR_KEY] = _load_component(
+            local_path, _PROCESSOR_KEY, flavor_config[_PROCESSOR_TYPE_KEY]
+        )
+
+    for component_key in flavor_config[_COMPONENTS_BINARY_KEY]:
+        component_type_key = f"{component_key}_type"
+        component_type = flavor_config[component_type_key]
+        conf[component_key] = _load_component(local_path, component_key, component_type)
+    return conf
+
+
+def _fetch_model_card(model_or_pipeline):
+    """
+    Attempts to retrieve the model card for the specified model architecture iff the
+    `huggingface_hub` library is installed. If a card cannot be found in the registry or
+    the library is not installed, returns None.
+    """
+    with contextlib.suppress(ImportError):
+        from transformers import Pipeline
+
+        hub = importlib.import_module("huggingface_hub")
+        model = (
+            model_or_pipeline.model
+            if isinstance(model_or_pipeline, Pipeline)
+            else model_or_pipeline[_MODEL_KEY]
+        )
+        return hub.ModelCard.load(model.name_or_path)
+
+
+def _build_pipeline_from_model_input(model, task: str):
+    """
+    Utility for generating a pipeline from component parts. If required components are not
+    specified, use the transformers library pipeline component validation to force raising an
+    exception. The underlying Exception thrown in transformers is verbose enough for diagnosis.
+    """
+    from transformers import pipeline
+
+    pipeline_config = {
+        "task": task,
+        **model,
+    }
+    try:
+        return pipeline(**pipeline_config)
+    except Exception as e:
+        raise MlflowException(
+            "The provided model configuration cannot be created as a Pipeline. "
+            "Please verify that all required and compatible components are "
+            "specified with the correct keys.",
+            error_code=INVALID_PARAMETER_VALUE,
+        ) from e
+
+
+def _record_pipeline_components(pipeline) -> Optional[Dict[str, Any]]:
```

Comment:
```suggestion
def _record_pipeline_components(pipeline) ->Dict[str, Any]:
```

It looks like this function always returns a dict.